### PR TITLE
Sharing variables with stages and jobs

### PR DESCRIPTION
**Pull request by wpschaub**
A troubleshooting journal with gotchas and a troubleshooting checklist.

### CODE REVIEW

**Comment by wpschaub** [comment]
@wsbctechnicalblog @erwsch1 @said-akram-wcbbc a summary blog of my pipeline (variables) troubleshooting afternoon. Would these types of journals add value to the blog? If not, I will move to wiki.

**Review by erwsch1 — DISMISSED** [review]

File: content/sharing-variables-with-stages-and-jobs.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    - bash: echo "##vso[task.setvariable variable=secretValue1;isOutput=true]BINGO-1!"
+      name: SetVariable1
+    - bash: echo "##vso[task.setvariable variable=secretValue2;isOutput=true]BINGO-2!"
+    - bash: echo "##vso[task.setvariable variable=secretValue3;isOutput=true]BINGO-3!"
+      name: SetVariable3
+    - task: DutchWorkzToolsAllVariables@1
+```
+
+
+The last task, which unfortunately only runs on a Windows-based agent, uses the [Display all variables](https://marketplace.visualstudio.com/items?itemName=dutchworkz.DisplayAllVariables) extension, which displays all variables that are available during your Build or Release. It is an exceptional troubleshooting companion and we find all three of our variables listed as individual variables, and as part of the VSTS_PUBLIC_VARIABLES collection.
```
Comment (said-akram-wcbbc) [inline content/sharing-variables-with-stages-and-jobs.md]: Hi @wpschaub, the "Display all variables" extension has been unpublished from Marketplace. I think we should avoid using it.
I use this simple script to log all the variables: `- bash: 'env | sort'`. It works on all agents.
Thanks

**Comment by wpschaub** [comment]
@said-akram-wcbbc I updated the post and sample. Thanks for the pointer!

@erwsch1 sorry, the update triggered a re-review.

**Review by erwsch1 — APPROVED** [review]

**Review by said-akram-wcbbc — APPROVED** [review]

**Comment by wpschaub** [comment]
@wsbctechnicalblog you can click the launch button.

**Pull request merged**